apps/wakunode2/wakunode2.nim

```diff
@@ -46,98 +49,157 @@ when defined(rln) or defined(rlnzerokit):
+  # if (pageCount > 0 and freelistCount > 0):
 
-      let resVacuum = sqliteDatabase.vacuum()
-      if resVacuum.isErr():
-        return err("failed to execute vacuum: " & resVacuum.error())
+  debug "starting sqlite database vacuuming"
 
-      debug "finished sqlite database vacuuming"
+  let resVacuum = db.vacuum()
+  if resVacuum.isErr():
+    return err("failed to execute vacuum: " & resVacuum.error)
 
-    sqliteDatabase.runMigrations(conf) 
+  debug "finished sqlite database vacuuming"
 
+proc performDbMigration(db: SqliteDatabase, migrationPath: string): SetupResult[void] =
+  ## Run migration scripts on persistent storage
+  debug "starting sqlite database migration"
+  let migrationRes = db.migrate(migrationPath)
+  if migrationRes.isErr():
+    return err("failed to execute migration scripts: " & migrationRes.error)
+
+  debug "finished sqlite database migration"
+
+
+const PeerPersistenceDbUrl = "sqlite://peers.db"
+
+proc setupPeerStorage(): SetupResult[Option[WakuPeerStorage]] =
+  let db = ?setupDatabaseConnection(PeerPersistenceDbUrl)
+  
+  ?performDbMigration(db.get(), migrationPath=MessageStoreMigrationPath)
 
-  if conf.persistPeers:
-    let res = WakuPeerStorage.new(sqliteDatabase)
+  let res = WakuPeerStorage.new(db.get())
+  if res.isErr():
+    return err("failed to init peer store" & res.error)
+  
+  ok(some(res.value))
+
+
+proc setupMessagesStore(db: Option[SqliteDatabase], storeCapacity: int = high(int)): SetupResult[MessageStore] =
+  if db.isSome():
+    debug "setting up sqlite-only message store"
+    let res = SqliteStore.init(db.get())
     if res.isErr():
-      warn "failed to init peer store", err = res.error
-      waku_node_errors.inc(labelValues = ["init_store_failure"])
-    else:
-      storeTuple.pStorage = res.value
-
-  if conf.persistMessages:
-    if conf.sqliteStore:
-      debug "setting up sqlite-only store"
-      let res = SqliteStore.init(sqliteDatabase)
-      if res.isErr():
-        warn "failed to init message store", err = res.error
-        waku_node_errors.inc(labelValues = ["init_store_failure"])
-      else:
-        storeTuple.mStorage = res.value
-    elif not sqliteDatabase.isNil():
-      debug "setting up dual message store"
-      let res = DualMessageStore.init(sqliteDatabase, conf.storeCapacity)
-      if res.isErr():
-        warn "failed to init message store", err = res.error
-        waku_node_errors.inc(labelValues = ["init_store_failure"])
-      else:
-        storeTuple.mStorage = res.value
-    else:
-      debug "setting up in-memory store"
-      storeTuple.mStorage = StoreQueueRef.new(conf.storeCapacity)
+      return err("failed to init sqlite message store: " & res.error)
+    
+    return ok(res.value)
+
+  else:
+    debug "setting up in-memory message store"
+    let store = StoreQueueRef.new(storeCapacity)
+    return ok(store)
+
+proc setupMessageStoreRetentionPolicy(retentionPolicy: string): SetupResult[Option[MessageRetentionPolicy]] =
```

Comment:
Should we use an optional enum for the retentionPolicy argument since we have only 2, or *no* retention policy at all?

Reply:
Can you elaborate on that? Where are you suggesting using the enum?

- Return type?
- Input argument?
- Config?

Reply:
in the input, sorry. `retentionPolicy: string`